backend/devices/bitbox/keystore.go

```diff
@@ -75,11 +77,18 @@ func (keystore *keystore) ExtendedPublicKey(
 
 // SignTransaction implements keystore.Keystore.
 func (keystore *keystore) SignTransaction(proposedTx coin.ProposedTransaction) error {
-	btcProposedTx, ok := proposedTx.(*btc.ProposedTransaction)
-	if !ok {
-		panic("only btc")
+	switch specificProposedTx := proposedTx.(type) {
+	case *btc.ProposedTransaction:
+		return keystore.SignBTCTransaction(specificProposedTx)
+	case *eth.TxProposal:
+		return keystore.SignETHTransaction(specificProposedTx)
+	default:
+		panic("unknown proposal type")
 	}
-	keystore.log.Info("Sign transaction")
+}
+
+func (keystore *keystore) SignBTCTransaction(btcProposedTx *btc.ProposedTransaction) error {
```

Comment:
Make method private. (And personal preference: Avoid forward reference in `SignTransaction`, i.e. move this method above `SignTransaction` – but we're not consistent anyway and I can live without such a change.)

Reply:
This is not C :P Done :)